### ISSUE TITLE
update the TODO example.

### ISSUE THREAD

**Issue by dogomayana** [rrousselGit/riverpod#3475]
class TodoList extends Notifier<List<Todo>> {
  @override
  List<Todo> build() => [
        const Todo(id: 'todo-0', description: 'Buy cookies'),
        const Todo(id: 'todo-1', description: 'Star Riverpod'),
        const Todo(id: 'todo-2', description: 'Have a walk'),
      ];

  void add(String description) {
    state = [
      ...state,
      Todo(
        id: _uuid.v4(),
        description: description,
      ),
    ];
  }
...
The "extend Notifier" throws a subclass error when build-runner run, this means, the "Notifier" is not the right notifier for that instance.

**This worked for me.**

![riverpod](https://github.com/rrousselGit/riverpod/assets/91336583/634357c6-e063-4dbd-b71c-5b1127e8ab00)
![riverpod1](https://github.com/rrousselGit/riverpod/assets/91336583/36ba00ec-df71-4601-8263-5800d5bf2ae3)

**In summary, substitute "Notifier" with "AutoDisposeNotifier".**

**Comment by rrousselGit** [comment]
The todo example is correct. You used autoDispose, when the example does not.

**Comment by dogomayana** [comment]
I didn't explicitly use the autoDispose. 
I copied the exact code from the TODO example and ran the build-runner which generated the autoDispose.

**Comment by rrousselGit** [comment]
You used `AutoDisposeNotifierProvider` in your code snippet, which isn't used anywhere in the documentation
